lib/fizzy/parser_expr.cpp

```diff
@@ -63,6 +63,36 @@ struct ControlFrame
     {}
 };
 
+enum class OperandStackType : uint8_t
+{
+    Unknown,
+    i32 = static_cast<uint8_t>(ValType::i32),
+    i64 = static_cast<uint8_t>(ValType::i64),
+    f32 = static_cast<uint8_t>(ValType::f32),
+    f64 = static_cast<uint8_t>(ValType::f64),
+};
+
+inline OperandStackType from_valtype(ValType valType) noexcept
+{
+    return static_cast<OperandStackType>(valType);
+}
+
+inline bool type_matches(OperandStackType actual_type, ValType expected_type) noexcept
+{
+    if (actual_type == OperandStackType::Unknown)
+        return true;
+
+    return static_cast<ValType>(actual_type) == expected_type;
+}
+
+inline bool type_matches(OperandStackType actual_type, OperandStackType expected_type) noexcept
```

Comment:
```suggestion
inline bool operator==(OperandStackType actual_type, OperandStackType expected_type) noexcept
```
?

Reply:
I didn't go with operators, because then we should probably also have `operator!=` for completeness, and for all pairs `OperansStackType`-`OperandStackType`, `OperandStackType`-`ValType`, `ValType`-`OperandStackType`, and unit tests for all of them...

Also it doesn't seem exacly like equivalence-kind of relation.

Reply:
I have similar doubts.